src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.html

```diff
@@ -150,7 +150,13 @@
                 <mat-menu #tableOptions="matMenu" xPosition="before" [hasBackdrop]="true">
                     <button mat-menu-item disabled="true">Details</button>
                     <mat-divider></mat-divider>
-                    <button mat-menu-item disabled="true">Cancel flow</button>
+                    <button
+                        (click)="cancelFlow(element.flowId)"
+                        mat-menu-item
+                        [disabled]="element.status !== FlowStatus.Waiting"
```

Comment:
It's okay to cancel running flow too. We should not allow cancelling Finished.

Reply:
Fixed